src/content/docs/en/upgrade-astro.mdx

```diff
@@ -0,0 +1,188 @@
+
+- [Upgrade to v4](/en/guides/upgrade-to/v4/)
+- [Upgrade to v3](/en/guides/upgrade-to/v3/)
+- [Upgrade to v2](/en/guides/upgrade-to/v2/)
+- [Upgrade to v1](/en/guides/upgrade-to/v1/)
+
+### Older docs (unmaintained)
+
+Documentation for older versions of Astro is not maintained, but is available as a static snapshot. Use these versions of docs if you are unable to upgrade your project, but still wish to consult guides and reference:
+
+- [unmaintained v3.6.3 snapshot](https://docs-git-v3-docs-unmaintained-astrodotbuild.vercel.app/)
+- [unmaintained v2.10.15 snapshot](https://deploy-preview-4405--astro-docs-2.netlify.app/en/getting-started/)
+
+
+## Semantic versioning
+
+Astro attempts to adhere as much as possible to [semantic versioning](https://semver.org/), which is a set of rules developers use to determine how to assign a version number to a release. Semantic version follows predictable to inform users of the kind of changes they can expect from one version to the next.
+
+Semantic versioning enforces a pattern of `X.Y.Z` for software version numbers. These values represent **major (X)**, **minor (Y)**, and **patch (Z)** updates.
+
+### Patch changes
+
+Patch changes are the least disruptive changes. They do not change the way you use Astro, and no change to your own code is required when you update.
+
+When Astro issues a "patch" version, the last number increases. (e.g. `astro@4.3.14` -> `astro@4.3.15`)
+
+Patches may be released for reasons such as:
+
+- Re-releases after a failed release.
+- Internal changes that do not change Astro’s functionality:
+  - refactors
+  - performance improvements
+  - increase or change in test coverage
+  - aligning with stated documentation and expected behavior
+- Improvements to logging and error messages.
+
+Patch changes also include bug fixes, even in cases where users were taking advantage of existing unintended or undesirable behavior.
+
+
+### Minor changes
+
+
+Minor releases primarily introduce new features and improvements that you may wish to try, but require no changes to your code. Some existing features may also be **deprecated** (marked for deletion in a future version while continuing to function) in a minor release, giving you the opportunity to prepare for their eventual removal.
+
+Minor releases include changes such as:
+
+- **Deprecations** of existing features/options.
+- Introduction of new functionalities.
+- Upgrade of minimal required Node.js version (minor or patch), e.g. from `v18.4.0` to `v18.7.0`.
+  :::note
+  We take these kind of changes very seriously, and approve them only if we think that a new version of Node.js can improve the DX of our users, and impact core functionalities.
```

Comment:
Discussion: Something I proposed before bumping the node version is perhaps to issue a warning from the CLI if they are using the soon-to-be-dropped nodejs versions. This way users will see warnings a couple minors before before we bump it. I'm not sure if we want to do this, but if you think this is a good idea, maybe we can note about this here. But if not (for now), I'm also fine 😄

Reply:
I don't mind the warning, although we have to consider that this is another "chore" to take into consideration between releases. If there's an automated way to do so, that would be great

Reply:
Yeah we'd have to remind ourselves if we do this, but given that we'd rarely do this unless we need to, I think it shouldn't take too much work. e.g. add the semver warnings, add a linear ticket to remind at a certain time, remove the semver warnings and update versions.